iris_ugrid/tests/unit/tests/synthetic_data_generator/test_synthetic_data.py

```diff
@@ -33,10 +31,13 @@ def tearDownClass(cls):
         # Destroy the temp directory.
         shutil.rmtree(cls.temp_dir)
 
-    def create_synthetic_testcube(self, **create_kwargs):
+    def create_synthetic_testcube(self, dataset_type, **create_kwargs):
 
-        file_path = create_file__xios_half_levels_faces(
-            temp_file_dir=self.temp_dir, dataset_name="mesh", **create_kwargs
+        file_path = create_file(
```

Comment:
If we make `create_file` generic, as suggested above, I think it will be too much fuss to test it in a "dataset-type neutral" way, as that would (ideally) use a minimal template file just for testing it.
Instead, pragmatically, I would base 'Test_create_file' on general properties of an actual useful template.
So I would take roughly what we have here, based on a specific template/dataset-type, and reduce to a minimum by removing "most" of the type-specific functionality tests.

However, we will *also* then want type-specific tests for each dataset-type and its template definition.
If we re-introduce a specific routine for each dataset-type, we should simply have a testclass for each of those.

I don't think it matters too much how you divide this up, as long as all the relevant code is exercised.

Reply:
The changes in b62b005f6ba69f98a48a0f9d1ec35028e39c3aba mean the testing structure can be fairly 'flat', and avoid this problem.

Reply:
Yes, I simply missed that, as the underlying routines are now private, we don't actually need separate tests :+1: